### DISCUSSION TITLE
PPM not stable.

### DISCUSSION THREAD

**Discussion by thiensu1122** [HefnySco/OBAL#9] [Q&A] [answered]
Hi HefnySco, 

I got into a problem with PPM. I follow your youtube tutorial about how to enable it and it works, but I don't know why it doesn't really stable as you can see in the video. 

https://drive.google.com/file/d/1bfaHCGg9F2aEwT9ZWQVBakmPM7W3v33x/view?usp=sharing

Can you give me advice on this?

Thank you

**Comment by HefnySco** [comment, marked as answer]
Sorry for late reply.
I got similar results with pwm. I believe it is a driver issue when measuring the pwm pulse.

**Reply by highfreq** [reply]
Is ppm problem going to be solved? Will it be possible to run just a single wire from receiver to board? I am using an Herelink and i can use only ppm, there aren't single channel cables.

**Comment by HefnySco** [comment]
I recommend either using the STM32 RCInput module ... it works great.

https://youtu.be/CYxhLD3E1F8

or using [ PWM/PPM Decoder 8 Channel ](https://www.aliexpress.com/item/1005001892137682.html?spm=a2g0o.order_list.order_list_main.16.5a6d18021L1d5F)

**Reply by highfreq** [reply]
Hello, thank you. Is there any documentation on how to use STM32 rcinput?

**Reply by DroneEngage** [reply]
please check [this](https://medium.com/@mohammad.hefny/ardupilot-linux-based-system-stm32-as-a-breakout-module-part-2-5db4ce7cfe3e
)   
![Screenshot from 2023-03-26 15-34-53](https://user-images.githubusercontent.com/118908305/227779377-3cdd58c3-c218-45d8-95a8-2b67bf97f438.png)

**Reply by highfreq** [reply]
Any chance you are going to write a decent documentation on how to implement STM32 support on obal?

**Reply by HefnySco** [reply]
I am currently in the process of working on it. While I have completed the LCD module, I have not yet tested the device as a whole.
